mpi/inner_script.sh

```diff
@@ -9,13 +9,16 @@ die () {
 BATCH_NNODES=$(flux resource list -n -o {nnodes})
 BATCH_NCORES=$(flux resource list -n -o {ncores})
 COMPILER=$1
-MPI=$2
+MPI=$(echo $2 | sed "s/\///")
 export NAME="$COMPILER"_"$MPI"
 
 test -n $COMPILER || die "COMPILER (argument 1) not set"
 test -n $MPI || die "MPI (argument 2) not set"
+if [[ $MPI == "openmpi5" ]]; then
+    module use -q /usr/global/tools/mpi/toss_4_x86_64_ib/modulefiles/Core/
+fi
```

Comment:
There should probably be a comment here about why this module use line is needed.

Should it also be a fatal error if `module use` fails?

Reply:
I don't _think_ `module use` throws any errors, or, at least, it's very easily fooled. However, if the module isn't loaded, openmpi/5 won't be found and that will kill the script. (lines 20-21)

```
(s=125,d=0) corona82 ~ $ mkdir junkdir
(s=125,d=0) corona82 ~ $ module use /g/g0/hobbs17/junkdir/
(s=125,d=0) corona82 ~ $ vim junkdir/hello.txt
(s=125,d=0) corona82 ~ $ module use /g/g0/hobbs17/junkdir/
(s=125,d=0) corona82 ~ $ module use /g/g0/hobbs17/junkdir/hello.txt
(s=125,d=0) corona82 ~ $ $?
bash: 0: command not found...

```